### PR TITLE
[Broker] Other old values are still valid when updating resource group

### PR DESCRIPTION
**Pull request by yuruguo**
### Motivation
Currently, the remaining old values will be invalid when we update some values of the resource group, it is as follows:
**step 1**
```
bin/pulsar-admin resourcegroups create --msg-publish-rate 100 --byte-publish-rate 1024 --msg-dispatch-rate 200 --byte-dispatch-rate 2048 rg-test-01
```
| option | value |
| ------ | ------ |
| msg-publish-rate | 100 |
| byte-publish-rate | 1024 |
| msg-dispatch-rate | 200 |
| byte-dispatch-rate | 2048 |

**step 2**
```
bin/pulsar-admin resourcegroups update --msg-publish-rate 150 rg-test-01
```
| option | value |
| ------ | ------ |
| msg-publish-rate | 150 |
| byte-publish-rate | -1 |
| msg-dispatch-rate | -1 |
| byte-dispatch-rate | -1 |

The purpose of this PR is to ensure other old values are still valid when updating resource group.
**NOTE**:  #12242 and #12251  are similar to this PR.

### Documentation
- [x] `no-need-doc`

### CODE REVIEW

**Comment by yuruguo** [comment]
@eolivelli @codelipenghui @hangc0276 @315157973 ptal

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ResourceGroupsBase.java [hunk 1]
```diff
@@ -63,10 +63,18 @@ protected void internalUpdateResourceGroup(String rgName, ResourceGroup rgConfig
             /*
              * assuming read-modify-write
              */
-            resourceGroup.setPublishRateInMsgs(rgConfig.getPublishRateInMsgs());
-            resourceGroup.setPublishRateInBytes(rgConfig.getPublishRateInBytes());
-            resourceGroup.setDispatchRateInMsgs(rgConfig.getDispatchRateInMsgs());
-            resourceGroup.setDispatchRateInBytes(rgConfig.getDispatchRateInBytes());
+            if (rgConfig.getPublishRateInMsgs() != -1) {
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ResourceGroupsBase.java:66]: What does `-1` means here? do not have a limit for publish rate? or `0` means no limit for publish rate?

Comment (yuruguo) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ResourceGroupsBase.java:66]: The default value of `rate` is `- 1`
https://github.com/apache/pulsar/blob/ef8120f0ae16d829036292dc0b50640bc515ae98/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdResourceGroups.java#L87-L106
https://github.com/apache/pulsar/blob/ef8120f0ae16d829036292dc0b50640bc515ae98/pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/PublishRate.java#L28-L37
which means not take effect or not limit.

**Review by eolivelli — APPROVED** [review]
Lgtm

**Comment by eolivelli** [comment]
Please note that this is kind of behavior change and we cannot cherry pick this to other released branches.

I also marked this as to be noted in the release notes.

Can you please advertise this change on dev@ 
This way more folks will be notified

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ResourceGroupsBase.java (hunk 1 above)
Comment (yuruguo) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ResourceGroupsBase.java:66]: @codelipenghui ptal again, thx!

**Comment by yuruguo** [comment]
> Please note that this is kind of behavior change and we cannot cherry pick this to other released branches.
> 
> I also marked this as to be noted in the release notes.
> 
> Can you please advertise this change on dev@ This way more folks will be notified

I have sent a notification email to @dev

**Comment by yuruguo** [comment]
@merlimat @freeznet PTAL, thx!

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ResourceGroupsBase.java (hunk 1 above)
Comment (codelipenghui) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ResourceGroupsBase.java:66]: Yes, this was why I am considering this part. If the existing resource group is 

```java
private int publishRateInMsgs = 10; 
private long publishRateInBytes = 10; 
private int dispatchRateInMsgs = 10; 
private long dispatchRateInBytes = 10;
```

And want to remove the limit for `dispatchRateInMsgs`, the new resource group should be:

```java
private int publishRateInMsgs = 10; 
private long publishRateInBytes = 10; 
private int dispatchRateInMsgs = -1; 
private long dispatchRateInBytes = 10;
```

But we can't apply `dispatchRateInMsgs = -1` in your fix right?

Comment (yuruguo) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ResourceGroupsBase.java:66]: Good point, you are right!
`-1` is an occupied value, we can‘t judge whether `-1` is the default value or  user-specified value, so maybe we should remove the `-1` and judge by `null`, redefine `ResourceGroup`
```
 public class ResourceGroup { 
     private int publishRateInMsgs = -1; 
     private long publishRateInBytes = -1; 
     private int dispatchRateInMsgs = -1; 
     private long dispatchRateInBytes = -1; 
 } 
```

**Comment by yuruguo** [comment]
pulsarbot run-failure-checks

**Comment by yuruguo** [comment]
pulsarbot run-failure-checks

**Comment by yuruguo** [comment]
@codelipenghui ptal thx

**Comment by yuruguo** [comment]
@codelipenghui @hangc0276 PTAL, thx!

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@Jason918 Plesae help review this PR, thanks.

**Review by hangc0276 — APPROVED** [review]

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdResourceGroups.java [hunk 2]
```diff
@@ -55,23 +55,23 @@ void run() throws PulsarAdminException {
         @Parameter(names = { "--msg-publish-rate",
                 "-mp" }, description = "message-publish-rate "
                 + "(default -1 will be overwrite if not passed)", required = false)
```
Comment (Jason918) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdResourceGroups.java:57]: > (default -1 will be overwrite if not passed)

Do we need to update this description ?

**Review by Jason918 — APPROVED** [review]
Overall LGTM.
Left one comment.

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdResourceGroups.java (hunk 2 above)
Comment (yuruguo) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdResourceGroups.java:57]: [Create](https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdResourceGroups.java#L51) command not need update but [Update](https://github.com/apache/pulsar/blob/master/pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdResourceGroups.java#L90) command need.

**Pull request merged**